src/cdsetool/query.py

```diff
@@ -72,7 +72,9 @@ def __fetch_features(self):
             total_results = res.get("properties", {}).get("totalResults")
             if total_results is not None:
                 self.total_results = total_results
-
+            odata = self.__fetch_odata_from_product_list(self.features)
```

Comment:
The `odata` value is never used outside of `__add_odata_to_features()`, so call  `self.__add_odata_to_features(self.__fetch_odata_from_product_list(self.fetures))` here and put the if-statement first in `__add_odata_to_features()` instead.